src/data_c.c4

```diff
@@ -439,6 +479,33 @@ SHMEM_PROF_DEF_CTX_PUT_N_SIGNAL_NBI(`mem')
     }                                                         \
   }
 
+#define SHMEM_DEF_IBPUT(STYPE,TYPE)                           \
+  void SHMEM_FUNCTION_ATTRIBUTES                              \
+  SHMEMX_FUNC_PROTOTYPE(STYPE##_ibput, TYPE *target,            \
+                       const TYPE *source, ptrdiff_t tst,     \
+                       ptrdiff_t sst, size_t bsize, size_t nblocks, int pe)  \
+    SHMEM_ERR_CHECK_INITIALIZED();                            \
+    SHMEM_ERR_CHECK_PE(pe);                                   \
+    SHMEM_ERR_CHECK_CTX(ctx);                                 \
+    SHMEM_ERR_CHECK_POSITIVE(tst);                            \
+    SHMEM_ERR_CHECK_POSITIVE(sst);                            \
+    SHMEM_ERR_CHECK_SYMMETRIC(target, sizeof(TYPE) * ((nblocks-1) * tst + 1)); \
+    SHMEM_ERR_CHECK_NULL(source, nblocks);                    \
+    SHMEM_ERR_CHECK_STRIDE_GTE_BSIZE(tst, bsize);             \
+    SHMEM_ERR_CHECK_STRIDE_GTE_BSIZE(sst, bsize);             \
+    SHMEM_ERR_CHECK_OVERLAP(target, source,                   \
+                   sizeof(TYPE) * ((nblocks-1) * tst + 1),     \
+                   sizeof(TYPE) * ((nblocks-1) * sst + 1), 0); \
+    size_t i = 0;                                             \
+    for ( ; nblocks > 0 ; --nblocks) {                        \
+      for (i = 0; i < bsize; i++) {                           \
+        shmem_internal_put_scalar(ctx, target++, source++,    \
+                                  sizeof(TYPE), pe);          \
```

Comment:
It took me a while to get it.. but, curious to know, why didn't you write it in the following way? What am I missing?
```
    for ( ; nblocks > 0 ; --nblocks) {                        \
      for (i = 0; i < bsize; i++) {                           \
        shmem_internal_put_scalar(ctx, target[i], source[i],    \
                                  sizeof(TYPE), pe);          \
      }                                                       \
      target += tst;                                  \
      source += sst;                                  \
    }                                                         \
  }
```

Reply:
From what I can tell, the two implementations are functionally the same, although, I agree that the approach you've provided is easier to understand, so I'll go ahead and make the change here and in the other applicable functions

Reply:
Went ahead and changed the implementation so the inner for-loop that runs for **bsize** iterations is no longer needed. Instead, for each iteration in the **nblocks** loop, the implementation now does a single put or get of size `bsize * sizeof(TYPE)`